Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,241 @@
+      .withServiceProviders([reflectionService])
+      .withLogger(self.serverLogger)
+      .bind(host: "127.0.0.1", port: 0)
+      .wait()
+    self.server = server
+
+    let channel = try GRPCChannelPool.with(
+      target: .hostAndPort("127.0.0.1", server.channel.localAddress!.port!),
+      transportSecurity: .plaintext,
+      eventLoopGroup: MultiThreadedEventLoopGroup.singleton
+    ) {
+      $0.backgroundActivityLogger = self.clientLogger
+    }
+    self.channel = channel
+  }
+
+  override func tearDown() {
+    if let channel = self.channel {
+      XCTAssertNoThrow(try channel.close().wait())
+    }
+    if let server = self.server {
+      XCTAssertNoThrow(try server.close().wait())
+    }
+
+    super.tearDown()
+  }
+
+  func testFileByFileName() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(.with {
+      $0.host = "127.0.0.1"
+      $0.fileByFilename = "bar.proto"
+    })
+    serviceReflectionInfo.requestStream.finish()
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(serializedData: (
+        message?.fileDescriptorResponse
+          .fileDescriptorProto[0]
+      )!)
+    XCTAssertEqual(receivedFileDescriptorProto.name, self.fileDescriptorProto!.name)
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.count,
+      self.fileDescriptorProto!.service.count
+    )
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.first!.method.first!.name,
+      self.fileDescriptorProto!.service.first!.method.first!.name
+    )
+    XCTAssertEqual(message?.fileDescriptorResponse.fileDescriptorProto.count, 5)
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(.with {
+      $0.host = "127.0.0.1"
+      $0.listServices = "services"
+    })
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }
+    XCTAssertEqual(receivedServices?.count, 5)
+    for serviceName in receivedServices! {
+      XCTAssertTrue(self.servicesNames.contains(serviceName))
+    }
+  }
+
+  func testReflectionServiceData() throws {
+    self.createFileDescriptorProto()
+    do {
+      let registry = try ReflectionServiceData(fileDescriptorProtos: self.protos)
```

Comment:
You should be testing each registry method in separate tests, it makes them easier to follow and debug when they fail.

Reply:
This is the test that checks the init, that's why it's checking both the dictionary and the service names list. Should these be done separately? I used the getters since the attributes are private

Reply:
Should I switch to internal as the access level for the attributes and not use the getters anymore?

Reply:
Should I switch to internal as the access level for the attributes and not use the getters anymore?

Reply:
> This is the test that checks the init, that's why it's checking both the dictionary and the service names list. Should these be done separately? I used the getters since the attributes are private

Yes. Typically you should start by testing the API of the type you're testing. There are two APIs which need testing here:
1. `getServices()`
2. `getfileDescriptorData()`

At the moment all the work to prepare values for these functions is done in `init`, but that's an implementation detail that could change.

Reply:
> Should I switch to internal as the access level for the attributes and not use the getters anymore?

Yes